test/integration/job/initHotKeysJob_test.re

```diff
@@ -139,12 +144,101 @@ let _ =
+          MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode();
+
           triggerDeleteHotKeyEvent();
 
           BuildComponentTool.buildSceneTree(TestTool.buildEmptyAppState())
           |> ReactTestTool.createSnapshotAndMatch;
         })
       );
+
+      describe("test bind focus hot-key", () => {
+        describe("if has currentScenetreeNode", () => {
+          describe("if the currentSceneTreeNode is scene gameObject", () =>
+            test(
+              "key down f, arcballCamera should set target to be (0.,0.,0.)",
+              () => {
+              MainEditorSceneTool.unsafeGetScene()
+              |> GameObjectTool.setCurrentSceneTreeNode;
+
+              let editorState = StateEditorService.getState();
+              let engineState = StateEngineService.unsafeGetState();
+
+              triggerFocusHotKeyEvent();
+
+              editorState
+              |> SceneViewEditorService.unsafeGetEditCamera
+              |. GameObjectComponentEngineService.unsafeGetArcballCameraControllerComponent(
+                   engineState,
+                 )
+              |. ArcballCameraEngineService.unsafeGetArcballCameraControllerTarget(
+                   engineState,
+                 )
+              |> expect == (0., 0., 0.);
+            })
+          );
+
+          describe("else", () =>
+            test(
+              "key down f, arcballCamera should set target to be currentSceneTreeNode position",
+              () => {
+                MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode();
+
+                let editorState = StateEditorService.getState();
+                let engineState = StateEngineService.unsafeGetState();
+                let targetPosition = (10.0, 20.0, 15.0);
+
+                let firstCube =
+                  engineState |> MainEditorSceneTool.getFirstCube;
+
+                let engineState =
+                  engineState
+                  |> GameObjectComponentEngineService.unsafeGetTransformComponent(
+                       firstCube,
+                     )
+                  |. TransformEngineService.setPosition(
+                       targetPosition,
+                       engineState,
+                     );
+
+                triggerFocusHotKeyEvent();
+
+                editorState
+                |> SceneViewEditorService.unsafeGetEditCamera
+                |. GameObjectComponentEngineService.unsafeGetArcballCameraControllerComponent(
+                     engineState,
+                   )
+                |. ArcballCameraEngineService.unsafeGetArcballCameraControllerTarget(
+                     engineState,
+                   )
+                |> expect == targetPosition;
+              },
+            )
+          );
+        });
+
+        describe("else", () =>
+          test(
+            "key down f, arcballCamera should set target to be (0.,0.,0.)", () => {
+            let editorState = StateEditorService.getState();
```

Comment:
TODO test: should:
1.clear current scene tree node;
2.set target to (1.0,2.0,3.0);
3.focus